app/code/Magento/Quote/Model/QuoteManagement.php

```diff
@@ -368,6 +386,13 @@ public function placeOrder($cartId, PaymentInterface $paymentMethod = null)
             $quote->setCustomerGroupId(\Magento\Customer\Api\Data\GroupInterface::NOT_LOGGED_IN_ID);
         }
 
+        $remoteAddress = $this->remoteAddress->getRemoteAddress();
+        if ($remoteAddress) {
+            $quote->setRemoteIp($remoteAddress);
+            $xForwardIp = $this->request->getServer('HTTP_X_FORWARDED_FOR');
+            $quote->setXForwardedFor($xForwardIp);
```

Comment:
Maybe we should check for emptiness here? Variable name looks strange, why not `$forwardedForIp` or something?

Reply:
The code is the same as in `\Magento\Checkout\Model\remoteAddress::getQuote`.
I checked the docblock of `\Magento\Framework\HTTP\PhpEnvironment\RemoteAddress::getRemoteAddress`. IMHO the `@return` type is wrong. It should be "@return int|bool|string".
What do you mean?

Reply:
It also possible to inline the code. The variable is not really needed.